istioctl/pkg/install/pre-check_test.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2019 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package install
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+
+	authorizationapi "k8s.io/api/authorization/v1beta1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/version"
+	"k8s.io/kubernetes/pkg/kubectl/genericclioptions/resource"
+)
+
+type mockClientExecPreCheckConfig struct {
+	namespace  string
+	version    *version.Info
+	authConfig *authorizationapi.SelfSubjectAccessReview
+}
+type testcase struct {
+	description       string
+	config            *mockClientExecPreCheckConfig
+	expectedException bool
+}
+
+var (
+	version1 = &version.Info{
```

Comment:
It would be easier to read if these were `version1_13` and `version1_8` instead of `version1` and `version2`.